src/main/java/camera/Camera.java

```diff
@@ -43,21 +67,105 @@ public Mat getLastFrame() {
             changed = true;
         } else {
             changed = false;
+            return lastFrame.clone();
         }
 
+        divideFrame(newFrame);
+
+        addActivities(newFrame);
+
         return lastFrame.clone();
     }
 
+    /**
+     * Add all activities for every part of the frame.
+     * @param newFrame The frame to get the activity from.
+     */
+    private void addActivities(final Mat newFrame) {
+        final int modulus = 5;
+
+        if (frameCounter % modulus == 0) {
+            for (int i = 0; i < FRAMES; i++) {
+                addActivity(frameParts.get(i), i, knns.get(i));
+            }
+        }
+
+        addActivity(newFrame, FRAMES, knns.get(FRAMES));
+        frameCounter++;
+    }
+
+    /**
+     * Method in which a frame get divided into frames equally large parts.
+     * frames should be a number where the sqrt is an integer
+     *
+     * @param frame the input frame
+     */
+    public void divideFrame(final Mat frame) {
+        for (int i = 0; i < FRAMES; i++) {
+            int sqrt = (int) Math.sqrt(FRAMES);
+            int midCol = frame.width() / sqrt;
+            int midRow = frame.height() / sqrt;
+
+            int col = midCol * (i % sqrt);
+            int row = (i * sqrt) / FRAMES * midRow;
+
+            Mat result = frame.colRange(col, col + midCol);
+            result = result.rowRange(row, row + midRow);
+            frameParts.set(i, result);
+        }
+    }
+
+    /**
+     * Add current activity to list.
+     * @param frame the current frame
+     * @param partNumber current part of the frame
+     * @param knn the backgroundsubstractor
+     */
+    public void addActivity(final Mat frame, final int partNumber,
+                            final BackgroundSubtractorKNN knn) {
+        Mat subtraction = new Mat();
+        knn.apply(frame, subtraction);
+        Scalar meanValues = Core.mean(subtraction);
+
+        double change = 0;
+        for (double v : meanValues.val) {
+            change += v;
+        }
+
+        final int minFrames = 30;
+        final double timeWindow = 1000.00;
+
+        // Only add the activity to the list when at least some frames are processed.
+        if (frameCounter > minFrames) {
+            if (firstTime == -1) {
+                firstTime = System.currentTimeMillis();
+            }
+
+            long currentTime = System.currentTimeMillis() - firstTime;
+
+            double[] tuple = {currentTime / timeWindow, change};
```

Comment:
ik ga dr van uit dat currentTime / timeWindow de tij d in seconden weergeeft, is dat ook zo?

Reply:
Dat is zo ja.